### PR TITLE
feat(movies): rename "title" field in db to "name"

### PR DESCRIPTION
**Pull request by soph-iest**
**Note:** This PR is definitely too complex, and should really be broken up into several PRs. To save some time, rather than rebasing/reverting and re-applying all the changes, @sjl2 and I spoke about this, and walked through how, and in what order, these PRs should have been broken up and made.

*What:** 
- Updates controller to save new movies to the database
- Changes the database to use "name" rather than "title"

 **Why:** To practice zero-downtime db maintenance and migration practices

**Details:** Migration was done step-by-step with code changes according to the order specified in Miyagi, and tested successfully at each step (but wasn't broken up into commits/PRs because I'm an idiot).

How this should have been broken up:

- PR 1: _feat(movies): adds movie controllers._  Files changed:
  - lib/plugins/features/movies/controller.js
  - lib/plugins/features/movies/index.js
  - test/plugins/features/movies/controller.test.js
-  test/plugins/features/movies/index.test.js

- PR 2: _feat(movies): adds name, drops non-null from title in movies db_.  *also deploys.* Files changed:
  - db/migrations/20190402105303_add_name_to_movies.js

- PR 3: _feat(movies): populates name in code, migration backfills db _.  *code merges before deploy.* Files changed:
  - db/migrations/20190402111549_backfill_movies_name.js
  - lib/models/movie.js
  - lib/plugins/features/movies/controller.js
  - test/plugins/features/movies/controller.test.js

- PR 4: _feat(movies): removes title saving from code and db_.  *code merges before deploy.* Files changed:
  - db/migrations/20190402111917_drop_title_from_movies.js
  - lib/models/movie.js
  - lib/plugins/features/movies/controller.js
  - test/plugins/features/movies/controller.test.js

### CODE REVIEW

**Comment by soph-iest** [comment]
Question: One of the changes I had to make to pass travis testing was to change "before-scripts" in .travis.yml to "before_scripts" with an underscore, because otherwise they weren't running. Is this something we can change on the boilerplate? Seems important.

**Comment by sjl2** [comment]
oof. yeah. That sounds important. I'll look into fixing that.

File: db/migrations/20190402111549_backfill_movies_name.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+exports.up = (Knex) => {
+  return Knex.raw('UPDATE movies SET name = title');
+};
+
+exports.down = (Knex, Promise) => {
```
Comment (sjl2) [inline db/migrations/20190402111549_backfill_movies_name.js]: super nit: now that we can use `async` `await`, can be as simple as 
```js
export.down = async () => undefined
```

File: db/migrations/20190402111917_drop_title_from_movies.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+exports.up = (Knex) => {
+  return Knex.schema.table('movies', (table) => {
```
Comment (sjl2) [inline db/migrations/20190402111917_drop_title_from_movies.js]: nit: would love to see these as `async`/`await`.

File: test/plugins/features/movies/index.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const Movies = require('../../../../lib/server');
+
+describe('movies integration', () => {
+
+  describe('create', () => {
+
+    it('creates a movie', () => {
+      return Movies.inject({
```
Comment (sjl2) [inline test/plugins/features/movies/index.test.js]: super nit: async await

File: .eslintrc [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
-  "extends": "lob",
+  "extends": "lob/migrations",
+  "parserOptions": {
+    "ecmaVersion": 2018
+  },
   "globals": {
     "expect": false
   }
-}
+}
```
Comment (sjl2) [inline .eslintrc]: super nit: looks like this is missing a `\n`.

File: test/plugins/features/movies/index.test.js (hunk 3 above)
Comment (sjl2) [reply test/plugins/features/movies/index.test.js]: I see this was resolved, but the change wasn't actually implemented. Is that a mistake?

File: lib/plugins/features/movies/controller.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const Movie = require('../../../models/movie');
+
+exports.create = async (payload) => {
+  const dbEntry = {
+    name: payload.title,
```
[the remark this answers is not in the document]
Comment (soph-iest) [reply lib/plugins/features/movies/controller.js]: Yeah, that's what I had, here and in a few other files. My bad for not documenting that.

File: test/plugins/features/movies/index.test.js (hunk 3 above)
Comment (soph-iest) [reply test/plugins/features/movies/index.test.js]: Oops, yep. Resolved for real now, I think.

**Comment by soph-iest** [comment]
Thank you both for reviewing this, I know this is totally crunch time for Postmasters but I really appreciate it ❤

**Pull request merged**